service/history/workflow/task_generator.go

```diff
@@ -614,31 +614,31 @@ func (r *TaskGeneratorImpl) GenerateHistoryReplicationTasks(
 	return nil
 }
 
-func (r *TaskGeneratorImpl) GenerateMigrationTasks() (tasks.Task, error) {
+func (r *TaskGeneratorImpl) GenerateMigrationTasks() (tasks.Task, int64, error) {
 	executionInfo := r.mutableState.GetExecutionInfo()
 	versionHistory, err := versionhistory.GetCurrentVersionHistory(executionInfo.GetVersionHistories())
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 	lastItem, err := versionhistory.GetLastVersionHistoryItem(versionHistory)
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
 	if r.mutableState.GetExecutionState().State == enumsspb.WORKFLOW_EXECUTION_STATE_COMPLETED {
 		return &tasks.SyncWorkflowStateTask{
 			// TaskID, VisibilityTimestamp is set by shard
 			WorkflowKey: r.mutableState.GetWorkflowKey(),
 			Version:     lastItem.GetVersion(),
-		}, nil
+		}, 1, nil
 	} else {
 		return &tasks.HistoryReplicationTask{
 			// TaskID, VisibilityTimestamp is set by shard
 			WorkflowKey:  r.mutableState.GetWorkflowKey(),
 			FirstEventID: executionInfo.LastFirstEventId,
 			NextEventID:  lastItem.GetEventId() + 1,
 			Version:      lastItem.GetVersion(),
```

Comment:
does it make sense to include `StateTransitionCount` as part of `HistoryReplicationTask`?

Reply:
no, state transition is basically "how many LWT / Tx" has this workflow caused so far

this metrics can be useful for predicting the # of LWT. 

do you have any use case for this metrics?

Reply:
this is a go feature, not a bug ...